app/grandchallenge/cases/image_builders/metaio_utils.py

```diff
@@ -0,0 +1,165 @@
+import SimpleITK._SimpleITK as _SimpleITK
+
+METAIO_IMAGE_TYPES = {
+    "MET_NONE": None,
+    "MET_ASCII_CHAR": None,
+    "MET_CHAR": SimpleITK.sitkInt8,
+    "MET_UCHAR": SimpleITK.sitkUInt8,
+    "MET_SHORT": SimpleITK.sitkInt16,
+    "MET_USHORT": SimpleITK.sitkUInt16,
+    "MET_INT": SimpleITK.sitkInt32,
+    "MET_UINT": SimpleITK.sitkUInt32,
+    "MET_LONG": SimpleITK.sitkInt64,
+    "MET_ULONG": SimpleITK.sitkUInt64,
+    "MET_LONG_LONG": None,
+    "MET_ULONG_LONG": None,
+    "MET_FLOAT": SimpleITK.sitkFloat32,
+    "MET_DOUBLE": SimpleITK.sitkFloat64,
+    "MET_STRING": None,
+    "MET_CHAR_ARRAY": SimpleITK.sitkVectorInt8,
+    "MET_UCHAR_ARRAY": SimpleITK.sitkVectorUInt8,
+    "MET_SHORT_ARRAY": SimpleITK.sitkVectorInt16,
+    "MET_USHORT_ARRAY": SimpleITK.sitkVectorUInt16,
+    "MET_INT_ARRAY": SimpleITK.sitkVectorInt32,
+    "MET_UINT_ARRAY": SimpleITK.sitkVectorUInt32,
+    "MET_LONG_ARRAY": SimpleITK.sitkVectorInt64,
+    "MET_ULONG_ARRAY": SimpleITK.sitkVectorUInt64,
+    "MET_LONG_LONG_ARRAY": None,
+    "MET_ULONG_LONG_ARRAY": None,
+    "MET_FLOAT_ARRAY": SimpleITK.sitkVectorFloat32,
+    "MET_DOUBLE_ARRAY": SimpleITK.sitkVectorFloat64,
+    "MET_FLOAT_MATRIX": None,
+    "MET_OTHER": None,
+}
+
+
+def parse_mh_header(filename: Path) -> Mapping[str, Union[str, None]]:
+    """
+    Attempts to parse the headers of an mhd file. This function must be
+    secure to safeguard against any untrusted uploaded file.
+
+    Parameters
+    ----------
+    filename
+
+    Returns
+    -------
+
+    Raises
+    ------
+    ValueError:
+        raised when the file contains problems making it impossible to
+        read
+    """
+
+    # attempt to limit number of read headers to prevent overflow attacks
+    read_line_limit = 10000
+
+    result = {}
+    with open(str(filename), "rb") as f:
+        bin_line = True
+        while bin_line is not None:
+            read_line_limit -= 1
+            if read_line_limit < 0:
+                raise ValueError("Files contains too many header lines")
+
+            bin_line = f.readline(10000)
+            if not bin_line:
+                bin_line = None
+                continue
+            if len(bin_line) >= 10000:
+                raise ValueError("Line length is too long")
+
+            try:
+                line = bin_line.decode("utf-8")
+            except UnicodeDecodeError:
+                raise ValueError("Header contains invalid UTF-8")
+            else:
+                # Clean line endings
+                line = line.rstrip("\n\r")
+                if line.strip():
+                    if "=" in line:
+                        key, value = line.split("=", 1)
+                        result[key.strip()] = value.strip()
+                    else:
+                        result[line.strip()] = None
+            if "ElementDataFile" in result:
+                break  # last parsed header...
+    return result
+
+
+def load_sitk_image_with_nd_support_from_headers(

```

Comment:
It would be nice to simplfy this function a little, there are several places where some functions can be split out to help with readability. Same with `parse_mh_header`. If you have no time can do it.

Reply:
I will have a look at the functions. `parse_mh_headers` was implemented entirely by Paul though, and I have only moved, not touched, his function.

Reply:
I know, but this is a good opportunity to do some cleaning, always good to leave the codebase in a better state than you found it ;-)

Reply:
Ah, I have no problems with cleaning it, but I was a bit curious as to why it wasn't picked up before by code climate?

Reply:
We weren't running it back then! I'm going to integrate all of the useful checks from code climate in there to flake8 so that they're available in development too, and we can remove the dependency on a 3rd party service. There is another function in the codebase with a complexity of :100:, so, this is not the worst...

Reply:
That'll be very useful, right now it is kind of a pain to work with.